below/ethtool/src/reader.rs

```diff
@@ -0,0 +1,232 @@
+use std::os::fd::{OwnedFd, AsRawFd};
+use std::str;
+
+use nix::errno::Errno;
+use nix::libc::ioctl;
+use nix::sys::socket::{socket, AddressFamily, SockFlag, SockType};
+
+use crate::{
+    ETHTOOL_GSSET_INFO,
+    ETHTOOL_GSTRINGS,
+    ETHTOOL_GSTATS,
+    ethtool_stringset_ETH_SS_STATS as ETH_SS_STATS,
+    ETH_GSTRING_LEN,
+};
+use crate::errors::Error;
+
+const ETH_GSTRINGS_LEN: usize = ETH_GSTRING_LEN as usize;
+const ETH_GSTATS_LEN: usize = 8;
+
+/// Maximum size of an interface name
+const IFNAME_MAX_SIZE: usize = 16;
+
+/// MAX_GSTRINGS maximum number of stats entries that ethtool can retrieve
+const MAX_GSTRINGS: usize = 8192;
+
+#[derive(Debug)]
+#[repr(C)]
+struct StringSetInfo {
+    cmd: u32,
+    reserved: u32,
+    mask: u32,
+    data: usize,
+}
```

Comment:
Isn't this defined by ethtool.h? Why redefine it?

Reply:
i tried, but i was having trouble figuring out how to work with the struct generated by bindgen
```rs
#[repr(C)]
#[derive(Debug)]
pub struct ethtool_sset_info {
    pub cmd: __u32,
    pub reserved: __u32,
    pub sset_mask: __u64,
    pub data: __IncompleteArrayField<__u32>,
}
```

not sure how to init `__IncompleteArrayField`, will try figuring it out.